biometricks/src/main/java/com/handstandsam/biometricks/Biometricks.kt

```diff
@@ -48,5 +62,230 @@ sealed class Biometricks {
         fun from(context: Context): Biometricks {
             return BiometricksHelper(context).type
         }
+
+        /**
+         * Wrapper around [BiometricPrompt.authenticate] which handles many tricky issues for you.
+         * It will:
+         * - Wait for the app to be focused if not currently, as the prompt will fail to show
+         * otherwise.
+         * - Allow you to show/hide a loading indicator for when there's a delay showing the prompt.
+         * - Give you information on if you need to show an error to the user or not.
+         * ```
+         * lifecycleScope.launch {
+         *     try {
+         *         val unlockedCryptObject = Biometricks.showPrompt(
+         *             this@MainActivity,
+         *             lockedCryptoObject,
+         *             BiometricPrompt.PromptInfo.Builder()
+         *                 .setTitle(...)
+         *                 .setNegativeButtonText(...)
+         *                 .build()
+         *         ) { showLoading -> progressBar.visibility = if (showLoading) View.VISIBLE else View.INVISIBLE }
+         *
+         *         // success
+         *     } catch (e: BiometricException) {
+         *         // failure
+         *         if (e.showShow) {
+         *             // show error to the user
+         *         }
+         *     }
+         * }
+         * ```
+         *
+         * @param activity The host activity.
+         * @param cryptoObject The [BiometricPrompt.CryptoObject] to unlock. This is always required
+         * as you many not get 'secure' biometrics if you don't include it.
+         * @param promptInfo The [BiometricPrompt.PromptInfo] to display in the prompt.
+         * @param showLoading Callback to show/hide a loading indicator for when there's a delay
+         * showing the prompt. This is only used on api 28 as after that it shows immediately.
+         */
+        @MainThread
+        suspend fun showPrompt(
```

Comment:
Best to keep everything in one class or split it out into a separate one? Was 50/50 on this.

Reply:
I like it all on one.  There are only a few public methods and we wanna really call them out.

Reply:
May wanna take some implementation detail out of that class like I did with `BiometricHelper`, but we can start here and make more concise later.